app/models/order_item.rb

```diff
@@ -35,22 +32,38 @@ def less!
     end
   end
 
+  def adjust_if_product_stock_changed!
+    max_quantity = product.stock
+    return if quantity_ordered <= max_quantity
+    update_column(:quantity_ordered, max_quantity)
+    errors[:product_stock] << "Quantity ordered was adjusted because not enough of this product was stuck."
```

Comment:
not enough of this product was stuck into our inventory?

okay, I'll fix it. >_>